numba/tests/test_np_randomgen.py

```diff
@@ -0,0 +1,79 @@
+import numba
+import numpy as np
+
+from numba import types
+from numba.tests.support import TestCase
+from numba.np.random.generator_methods import get_proper_func
+from numpy.random import MT19937, Generator
+from numpy.testing import assert_equal, assert_array_equal, assert_raises
+
+
+def test_proper_func_provider():
+    def test_32bit_func():
+        return 32
+
+    def test_64bit_func():
+        return 64
+
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 np.float64)[0](), 64)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 np.float32)[0](), 32)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 types.float64)[0](), 64)
+    assert_equal(get_proper_func(test_32bit_func, test_64bit_func,
+                                 types.float32)[0](), 32)
```

Comment:
Numba uses the `unittest.TestCase` method based assertions and those methods available from `numba.tests.support.TestCase` which extends `unittest.TestCase`. Suggest using `self.assertEqual(...)` here and updating similar throughout? To get an array bitwise equality assertion I think `self.assertPreciseEqual(..., ulps=0)` should work.